sql/expression/comparison.go

```diff
@@ -54,8 +54,12 @@ func (c *comparison) Compare(ctx *sql.Context, row sql.Row) (int, error) {
 		return 0, err
 	}
 
-	if left == nil || right == nil {
+	if left == nil && right == nil {
 		return 0, ErrNilOperand.New()
+	} else if left == nil {
+		return 1, ErrNilOperand.New()
```

Comment:
Does it matter to return a correct result when you're also returning an error?

Reply:
This isn't very go-nuts

Reply:
How do you say idiomatic in golang

Reply:
Anyway, a better way to do this would be to override this for the NullSafeEquals type

Reply:
I should mention that this method isn't used outside this package, so you can define the semantics that make sense for nullSafeEquals